src/alert.cpp

```diff
@@ -1915,24 +1915,30 @@ namespace libtorrent
 		, std::vector<tcp::endpoint> const& peers)
 		: info_hash(ih)
 		, m_alloc(alloc)
-		, m_num_peers(int(peers.size()))
+		, m_num_peers(aux::numeric_cast<int>(peers.size()))
 	{
-		std::size_t total_size = peers.size(); // num bytes for sizes
+#if TORRENT_USE_IPV6
+		int total_size = m_num_peers; // num bytes for sizes
 		for (auto const& endp : peers)
 		{
-			total_size += endp.size();
+			total_size += endp.protocol() == tcp::v4() ? 6 : 18;
 		}
+#else
+		int total_size = m_num_peers * 6;
+#endif
 
-		m_peers_idx = alloc.allocate(int(total_size));
+		m_peers_idx = alloc.allocate(total_size);
 
-		char *ptr = alloc.ptr(m_peers_idx);
+		char* ptr = alloc.ptr(m_peers_idx);
 		for (auto const& endp : peers)
 		{
-			std::size_t const size = endp.size();
-			TORRENT_ASSERT(size < 0x100);
-			detail::write_uint8(size, ptr);
-			std::memcpy(ptr, endp.data(), size);
-			ptr += size;
+#if TORRENT_USE_IPV6
+			int const size = endp.protocol() == tcp::v4() ? 6 : 18;
+			detail::write_int8(size, ptr);
```

Comment:
I was suggesting having separate allocations for IPv4 and IPv6, to avoid having the size. it would be more compact